modules/pel/peltool/parse_user_data.py

```diff
@@ -40,25 +40,35 @@ def parse(self) -> str:
         else:
             value = self.parseCustom()
 
+        # Catch if value is None, otherwise python crashes
+        if value == None:
+            value = ("PROBLEM with UNRECOGNIZED value from parser creatorID={} compID={} subType={} version={}"
+                     .format(self.creatorID, hex(self.compID), self.subType, self.version))
+            return json.dumps(value)
```

Comment:
you still aren't returning the hexdump of the original data...

Could something like the following to get both:
output['Error'] = "Plugin returned a value of None..."
mv = memoryview(self.data)
output['Original Data'] = hexdump(mv)
return json.dumps(output)

Reply:
The problem I was solving was that there was NO data, so the message is the ONLY available output.

Reply:
there could be other cases where there is valid data passed in but the parser still returns None due to other reasons.

With the check you added on line 55 I don't think we'd get here with empty data again anyway.

Reply:
I think we still need to do the hexdump here because if a plugin returns None when the data isn't empty it wouldn't print any dadta.

Reply:
Help clarify something for me ... self.data is being processed outside this logic in parse or parseCustom, for the parse leg in the normal case the self.data comes back within the value parm, correct ?

Or is the statement that similarly to the new logic added in parseCustom, we need to do the same handling as was added under the Exception flow ?

Reply:
Done